### PR TITLE
fixes the case when output file is in working directory

### PR DESCRIPTION
**Pull request by mich-sanna**
bug fix

## Description
Fixes WriteXML when the directory field is an empty string

### CODE REVIEW

File: src/python/python/xml.py [hunk 1]
```diff
@@ -142,10 +142,14 @@ def set_filename(self, name):
...
 
         self.directory, main_file = os.path.split(name)
-        base_name = os.path.splitext(main_file)[0] # Remove the extension
+        base_name = os.path.splitext(main_file)[0]  # Remove the extension
 
-        if not os.path.isdir(self.directory):
-            os.makedirs(self.directory)
+        if len(self.directory) == 0:
+            self.directory = '.'
+
```
Comment (Speierers) [inline src/python/python/xml.py:149]: Remove this extra empty line

**Review by Speierers — COMMENTED** [review]
Thanks for this PR. Please squash those commits into a single one and I will then merge this onto `master`.

**Comment by mich-sanna** [comment]
squashed, rebased, and removed empty line after `self.directory = '.'`

thanks!

File: src/python/python/xml.py (hunk 1 above)
Comment (mich-sanna) [reply src/python/python/xml.py]: iI removed line 149

**Comment by Speierers** [comment]
Thanks 🚀

**Pull request merged**